src/sign_dlink_ru.c

```diff
@@ -183,15 +187,27 @@ FILE* make_out_file(char* filename) {
  * Sign the firmware file after all of our checks have completed
  */
 void sign_firmware(char* filename, char** salts, int num_salts) {
-	int i;
+	int i = 0;
 	salt_t salt;
 	FILE* f = make_out_file(filename);
+	
+	char* version_suffix = "c0ffeef0rge";
+			
 
+	if (strcmp(salts[0],"-v") == 0) {
+		init_salt(&salt, salts[1]);
```

Comment:
This line doesn't do anything, and is a memory leak, as the version suffix isn't used as a signature prefix.

Reply:
will look into